flink-streaming-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/LegacyFetcher.java

```diff
@@ -199,8 +200,10 @@ public void close() {
 			
 			FetchPartition[] partitions = partitionsList.toArray(new FetchPartition[partitionsList.size()]);
 
+			final KeyedDeserializationSchema<T> clonedDeserializer = InstantiationUtil.clone(deserializer, Thread.currentThread().getContextClassLoader());
```

Comment:
This classloader will not necessarily be able to deserialize the object. See comment above...